services/121-service/src/payments/reconciliation/intersolve-voucher-reconciliation/intersolve-voucher-reconciliation.service.ts

```diff
@@ -18,11 +18,45 @@ export class IntersolveVoucherReconciliationService {
     private readonly intersolveVoucherScopedRepository: ScopedRepository<IntersolveVoucherEntity>,
   ) {}
 
+  private async updateVouchersWithErrorHandling(
+    vouchers: IntersolveVoucherEntity[],
+    programId: number,
+    credentials: any,
+    errorMessagePrefix: string,
+    incrementBeforeTry: boolean,
+  ): Promise<number> {
+    let updatedCount = 0;
+    const errorVoucherIds: number[] = [];
+    for await (const voucher of vouchers) {
+      if (incrementBeforeTry) {
+        updatedCount++;
+      }
+      try {
+        await this.intersolveVoucherService.getAndUpdateBalance(
+          voucher,
+          programId,
+          credentials,
+        );
+        if (!incrementBeforeTry) {
+          updatedCount++;
+        }
+      } catch (error) {
+        errorVoucherIds.push(voucher.id);
+        if (errorVoucherIds.length >= 10) {
+          throw new Error(
+            `${errorMessagePrefix} ${errorVoucherIds.join(', ')}. Failing job to prevent excessive API calls.`,
+          );
+        }
+      }
+    }
+    return updatedCount;
+  }
+
   public async getAndUpdateBalancesForProgram(
     programId: number,
     jobName: IntersolveVoucherJobName,
   ): Promise<void> {
-    if (jobName === IntersolveVoucherJobName.getLastestVoucherBalance) {
+    if (jobName === IntersolveVoucherJobName.getLatestVoucherBalance) {
```

Comment:
nitpick - doing an early return here undents the rest of the code in this function by one level